src/core/pages/MarkdownMedia.tsx

```diff
@@ -0,0 +1,73 @@
+import { configRepo } from '~core/config';
+import {
+  parseMediaParams,
+  isYoutubeUrl,
+  getYoutubeEmbedUrl,
+  isExternalLink,
+} from './linkUtils';
+
+export function buildAssetUrl(asset: string) {
+  return `${configRepo.get().apiGateway}/apps/${configRepo.get().id}/assets/${asset}`;
+}
+
+/**
+ * Markdown media component that handles both images and embeds
+ *
+ * @example
+ * ```md
+ * Regular image:
+ * ![Alt text](path/to/image.jpg)
+ *
+ * Image with dimensions:
+ * ![Alt text](path/to/image.jpg::800,600)
+ *
+ * YouTube embed (default 560x315):
+ * ![Video title](https://youtube.com/watch?v=xyz)
+ *
+ * YouTube with custom params:
+ * ![Video title](https://youtube.com/watch?v=xyz::800,450,1)
+ * ```
+ */
+export function MarkdownMedia({
+  title,
+  alt,
+  src,
+}: {
+  title: string;
```

Comment:
The double expanded lists of properties and their types look difficult to read. Can we separately determine the props type above?